aspnetcore/performance/caching/memory.md

```diff
@@ -51,7 +51,10 @@ Use `System.Runtime.Caching`/`MemoryCache` as a compatibility bridge when portin
 
 ## Using IMemoryCache
 
-In-memory caching is a *service* that's referenced from your app using [Dependency Injection](../../fundamentals/dependency-injection.md). Call `AddMemoryCache` in `ConfigureServices`:
+> [!WARNING]
+> Using a *shared* memory cache from [Dependency Injection](xref:fundamentals/dependency-injection) and calling `SetSize`, `Size`, or `SizeLimit` to limit cache size can cause the app to fail. When a size limit is set on a cache, all entries must specify a size when being added. This can lead to issues since developers may not have full control on what uses the shared cache. For example, Entity Framework Core uses the shared cache and does not specify a size. If an app sets a cache size limit and uses EF Core, the app throw's an `InvalidOperationException`.
```

Comment:
"the app throw's an" -> "the app will throw an"

Reply:
I'd also add a bit about a proper solution in this case (construct your own cache singleton), and link to the sample below (`MyMemoryCache`) which implements it.

Reply:
> "the app throw's an" -> "the app will throw an"

Per MS style guide, avoid future tense - it doesn't MT (machine translate) well.

> add a bit about a proper solution

Added, please review

Reply:
It should be "throws" rather than "throw's" then, I think. It's not possessive.